### ISSUE TITLE
ngModal not working for input type checkbox. 

### ISSUE THREAD

**Issue by panthera** [angular/angular#28225]
# 🐞 bug report

### Affected Package
 @angular/forms

### Is this a regression?

Don't know

### Description

Doesn't seem to work when you are using the checkbox inside an ngFor, while repeating an array of objects like [{"checked":true},{"checked":false}]

##Repro

https://stackblitz.com/edit/angular-3orv4w

## 🔥 Exception or Error

No errors

## 🌍  Your Environment

**Angular Version:**
<pre><code>
Angular CLI: 7.1.1
Node: 10.14.2
OS: win32 x64
Angular: 7.1.1
... animations, cdk, cli, common, compiler, compiler-cli, core
... forms, language-service, material, platform-browser
... platform-browser-dynamic, router

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.11.1
@angular-devkit/build-angular     0.12.1
@angular-devkit/build-optimizer   0.12.1
@angular-devkit/build-webpack     0.12.1
@angular-devkit/core              7.1.1
@angular-devkit/schematics        7.1.1
@angular/pwa                      0.11.2
@angular/service-worker           7.1.2
@ngtools/webpack                  7.2.1
@schematics/angular               7.1.1
@schematics/update                0.11.1
rxjs                              6.3.3
typescript                        3.1.6
webpack                           4.23.1

</code></pre>

**Comment by trotyl** [comment]
Please provide a minimal reproduction via https://stackblitz.com/. Also, a working version number need to be specified if you want to mark it as regression.

**Comment by panthera** [comment]
@trotyl Updated the reproduction link

**Comment by trotyl** [comment]
Depending on event handler ordering on same element is invalid, `ngModel` (actually its `ControlValueAccessor`) is also powered by the `change` event (or its equivalent), and you must not rely the order of these two event handler.

That's say, one should only listen to `ngModelChange`, which is guaranteed to be run after `ngModel`'s internal listeners, which has been discussed at https://github.com/angular/angular/issues/9275.

Also, changing back the model binding value to its previous result would break change detection, which is duplicate of https://github.com/angular/angular/issues/13063.

If you want to take control over the view-to-model update process, you'll need to define your own `ControlValueAccessor`, or subscribing to https://github.com/angular/angular/issues/3009 for updates.

**Comment by panthera** [comment]
Used the temporary workaround to unblock myself with the help of the following approach

[](https://stackoverflow.com/questions/54250115/ngmodal-not-working-for-input-type-checkbox-inside-ngfor)

**Comment by kara** [comment]
@panthera Thanks for filing the issue. As @troytl mentions, there are a few existing issues that already document this use case, so I'm closing this issue as a dupe.
